src/Controls/src/Build.Tasks/NodeILExtensions.cs

```diff
@@ -429,10 +429,6 @@ static Instruction PushParsedEnum(XamlCache cache, TypeReference enumRef, string
 
 		public static IEnumerable<Instruction> PushXmlLineInfo(this INode node, ILContext context)
 		{
-			if (context.ValidateOnly)
-			{
-				yield break;
-			}
```

Comment:
Is there a reason for reverting https://github.com/dotnet/maui/pull/24493 ?

Reply:
Without this change, the test doesn't work (the service provider isn't instantiated at all so I cannot inspect the code that we generate). I'm considering letting the tests pass and then remove the test + the changes in this file and keep just the fix itself to avoid perf regressions (cc @jonathanpeppers).